tests/testutils/test_lint_module_output_update.py

```diff
@@ -0,0 +1,72 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+
+# pylint: disable=redefined-outer-name
+import sys
+from pathlib import Path
+from typing import Callable, Tuple
+
+import pytest
+
+from pylint.testutils import FunctionalTestFile
```

Comment:
Shall we add an test to see if the deprecation warning triggers correctly when doing this import?
I think it might be better to add this in a `with pytest.warns` guard anyway as we will otherwise always trigger the warning ourselves.

Reply:
I'd like to do that but I don't know what is triggering the deprecation warning, it's supposed to not ever be imported.

Reply:
According to test output it is `tests/testutils/test_output_line.py`

See:
```shell
tests/testutils/test_output_line.py::test_output_line_from_csv_error
124
  /home/runner/work/pylint/pylint/pylint/testutils/output_line.py:125: DeprecationWarning: In pylint 3.0 functional tests expected output should always include the expected confidence level, expected end_line and expected end_column. An OutputLine should thus have a length of 8.
125
    DeprecationWarning,
```

Reply:
What happens if somebody uses the current way to import it? Doesn't that trigger the deprecation?

Reply:
Well there's a warning when we launch the test at the beginning not when we re-import later in the test:
```python
    with pytest.warns(DeprecationWarning) as records:
        from pylint.testutils.functional_test_file import FunctionalTestFile
```
Result in :        
```
        with pytest.warns(DeprecationWarning) as records:
>           from pylint.testutils.functional_test_file import FunctionalTestFile
E           Failed: DID NOT WARN. No warnings of type (<class 'DeprecationWarning'>,) was emitted. The list of emitted warnings is: [].

tests/testutils/test_functional_test_file.py:5: Failed
============================================================================================ warnings summary =============================================================================================
pylint/testutils/functional_test_file.py:18
  /home/pierre/pylint/pylint/testutils/functional_test_file.py:18: DeprecationWarning: 'pylint.testutils.functional_test_file' will be accessible from the 'pylint.testutils.functional' namespace in pylint 3.0.

```
Notice how the deprecation Warning is taunting us at the bottom 😄 

Somehow there must be an import from this file that I can't seem to find.